pkg/module/blob/blob.go

```diff
@@ -0,0 +1,315 @@
+package blob
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	pathutil "path"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/mandelsoft/vfs/pkg/vfs"
+	"github.com/opencontainers/go-digest"
+)
+
+// MediaTypeTar defines the media type for a tarred file
+const MediaTypeTar = "application/x-tar"
+
+// MediaTypeGZip defines the media type for a gzipped file
+const MediaTypeGZip = "application/gzip"
+
+// MediaTypeOctetStream is the media type for any binary data.
+const MediaTypeOctetStream = "application/octet-stream"
+
+// Output is the output if read BlobInput.
+type Output struct {
+	Digest string
+	Size   int64
+	Reader io.ReadCloser
+}
+
+type InputType string
+
+const (
+	FileInputType = "file"
+	DirInputType  = "dir"
+)
+
+// Input defines a local resource input that should be added to the component descriptor and
+// to the resource's access.
+type Input struct {
+	// Type defines the input type of the blob to be added.
+	// Note that a input blob of type "dir" is automatically tarred.
+	Type InputType `json:"type"`
+	// MediaType is the mediatype of the defined file that is also added to the oci layer.
```

Comment:
```suggestion
	// MediaType is the media type of the defined file that is also added to the OCI layer.
```
You used upper case for OCI everywhere else